packages/core/src/lib/Uuid.ts

```diff
@@ -2,6 +2,17 @@ import { randomUUID } from "crypto"
 
 export type Uuid<T> = string & { __brand: T }
 
+// permissive uuid regex that allows non-standard uuids
+const regex = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/
+
+function parse<T>(str: string): Uuid<T> {
+  const lower = str.toLowerCase()
+  if (!regex.test(str)) {
```

Comment:
Did you intend to test `lower`, or should L9 be inlined?
Could this create function become a gate that canonicalises to 'N` format (i.e. no dashes/braces etc, lowercase) ?